utils/rr_graph_walk/rr_graph_walk.py

```diff
@@ -0,0 +1,389 @@
+            if context.depth < len(route):
+                route = route[:context.depth]
+
+            # Add current node to the route
+            route.append(context.node_id)
+
+            # Add all nodes that can be reached from this one through edges
+            is_leaf = True
+
+            # Walking forward
+            if walk_direction > 0:
+                if context.node_id in self.edges_from.keys():
+                    for dst in self.edges_from[context.node_id]:
+                        # We haven't visited the target
+                        if dst in walk_nodes:
+                            stack.append(
+                                RoutingGraph.WalkContext(
+                                    dst, context.depth + 1
+                                )
+                            )
+                            walk_nodes.remove(dst)
+                            is_leaf = False
+
+            # Walking backward
+            if walk_direction < 0:
+                if context.node_id in self.edges_to.keys():
+                    for src in self.edges_to[context.node_id]:
+                        # We haven't visited the target
+                        if src in walk_nodes:
+                            stack.append(
+                                RoutingGraph.WalkContext(
+                                    src, context.depth + 1
+                                )
+                            )
+                            walk_nodes.remove(src)
+                            is_leaf = False
+
+            # We are in a leaf node
+            if is_leaf:
+
+                # Check the route
+                if not self.verify_route(route, walk_direction):
+                    raise RuntimeError("The route was invalid!")
+
+                # Invoke the callback
+                res = route_callback(self, route)
+
+                # Stop walk
+                if res != True:
+                    return
+
+
+# =============================================================================
+
+
+def print_route(graph, route):
+    """
+    Print given route with detailed node informations
+    """
+
+    for id in route:
+        sys.stdout.write("%s, " % graph.node_to_string(id))
+    sys.stdout.write("\n")
+
+
+def save_route(route, fp):
+    """
+    Save the route to a file (only node indices)
+    """
+
+    for id in route:
+        fp.write("%d," % id)
+
+    fp.write("\n")
+    fp.flush()
+
+
+# =============================================================================
+
+
+def main():
+
+    # Parse arguments
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+
+    parser.add_argument(
+        "--rr_graph", type=str, required=True, help="Routing graph XML file"
+    )
+    parser.add_argument(
+        "-s", type=int, required=True, help="Start node id (required)"
+    )
+    parser.add_argument(
+        "-e",
```

Comment:
Maybe it is worth to expand the name of this argument (e.g. `-end_inode`)? (also for `-s`)

Reply:
Yep, good idea. Done it.

Reply:
You need to update the ``args.s``  -> ``args.start_inode``, etc.  The script does not work right now.

Reply:
Fixed.